rasa/shared/utils/validation.py

```diff
@@ -289,3 +289,48 @@ def validate_training_data_format_version(
         docs=DOCS_URL_TRAINING_DATA,
     )
     return False
+
+
+def validate_yaml_with_jsonschema(
+    yaml_file_content: Text, schema_path: Text, package_name: Text = PACKAGE_NAME
+) -> None:
+    """Validate data format.
+
+    Args:
+        yaml_file_content: the content of the yaml file to be validated
+        schema_path: the schema of the yaml file
+        package_name: the name of the package the schema is located in. defaults
+            to `rasa`.
+
+    Raises:
+        YamlSyntaxException: if the yaml file is not valid.
+        SchemaValidationError: if validation fails.
+    """
+    from jsonschema import validate, ValidationError
+    from ruamel.yaml import YAMLError
+    import pkg_resources
+
+    schema_file = pkg_resources.resource_filename(package_name, schema_path)
+    schema_content = rasa.shared.utils.io.read_yaml_file(schema_file)
+
+    try:
+        # we need "rt" since
+        # it will add meta information to the parsed output. this meta information
+        # will include e.g. at which line an object was parsed. this is very
+        # helpful when we validate files later on and want to point the user to the
+        # right line
+        source_data = rasa.shared.utils.io.read_yaml(
+            yaml_file_content, reader_type=["safe", "rt"]
+        )
+    except (YAMLError, DuplicateKeyError) as e:
+        raise YamlSyntaxException(underlying_yaml_exception=e)
+
+    try:
+        validate(source_data, schema_content)
+    except ValidationError as error:
+        error.message += (
+            f". Failed to validate data, make sure your data "
+            f"is valid. For more information about the format visit "
+            f"{DOCS_URL_TRAINING_DATA}."
```

Comment:
We'll need to check if this is the correct url to direct to + if not, if we need to add a specific section in the docs PR to describe the format.

Reply:
That is the case https://12795--rasahq-docs-rasa-v2.netlify.app/docs/rasa/training-data-format does not exist (at least on docs-dm2). The whole [docs constants](https://github.com/RasaHQ/rasa/blob/32c24348151180f618bba2b436174b707c0a03ed/rasa/shared/constants.py#L6) need to be updated then (I will make a ticket for that)